pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -317,21 +365,68 @@ func (rd *restoreDataProcessor) openSSTs(
 		return nil
 	}
 
+	hashSpan := func(span roachpb.Span) (uint32, error) {
+		hash := fnv.New32a()
+		_, err := hash.Write(span.Key)
+		if err != nil {
+			return 0, err
+		}
+
+		return hash.Sum32() % uint32(rd.numWorkers), nil
+	}
+
 	log.VEventf(ctx, 1 /* level */, "ingesting span [%s-%s)", entry.Span.Key, entry.Span.EndKey)
 
-	storeFiles := make([]storageccl.StoreFile, 0, len(EntryFiles{}))
-	for _, file := range entry.Files {
+	storeFiles := make([]storageccl.StoreFile, 0, len(entry.Files))
+	var bytesPerFile int64
+	if rd.spec.Encryption != nil {
+		bytesPerFile = sstReaderEncryptedMemBytesPerFile
+	} else {
+		bytesPerFile = sstReaderMemBytesPerFile
+	}
+
+	for idx := 0; idx < len(entry.Files); {
+		file := entry.Files[idx]
 		log.VEventf(ctx, 2, "import file %s which starts at %s", file.Path, entry.Span.Key)
 
+		if err := rd.mem.Grow(ctx, bytesPerFile); err != nil {
```

Comment:
Separately, how would you feel about un-indenting some of this, i.e. `if err :=...; err == nil { continue}` Think that might be more readable? or would the `err == nil` be too surprising?

Reply:
I think this is okay for now to keep `err != nil` as the conditional. We have large err blocks elsewhere too.